tests/integration-tests/tests/dcv/test_dcv.py

```diff
@@ -0,0 +1,79 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import boto3
+import pytest
+
+from assertpy import assert_that
+from remote_command_executor import RemoteCommandExecutor
+
+
+def _setup_cluster(clusters_factory, pcluster_config_reader):
+    cluster_config = pcluster_config_reader()
+    cluster = clusters_factory(cluster_config)
+    remote_command_executor = RemoteCommandExecutor(cluster)
+    return remote_command_executor
+
+
+# This mini-test is to make sure that dcv_authenticator blocks non valid sessions
+def _assert_dcv_authenticator_blocking(remote_command_executor):
+    assert_that(
+        remote_command_executor.run_remote_command(
+            "curl -s -X GET -G http://localhost:8444  -d action=requestToken -d authUser=centos -d sessionID=francescoFixes"
+        ).stdout
+    ).is_equal_to("The given session for the user does not exists")
+
+
+def _assert_dcv_working(external_authenticator_port, remote_command_executor):
+    output = remote_command_executor.run_remote_command(
+        "/opt/parallelcluster/scripts/pcluster_dcv_connect.sh /shared"
+    ).stdout
+    session_id, port, session_token = output.split()
+    assert_that(
+        remote_command_executor.run_remote_command(
+            f"curl -s -k http://localhost:{external_authenticator_port} "
+            f"-d sessionId={session_id} -d authenticationToken={session_token} -d clientAddr=someIp"
+        ).stdout
+    ).is_equal_to('<auth result="yes"><username>centos</username></auth>')
+
+
+def _assert_security_group(expected, port, cluster, region):
+    security_group_id = cluster.cfn_resources.get("MasterSecurityGroup")
+    client = boto3.client("ec2", region_name=region)
+    response = client.describe_security_groups(GroupIds=[security_group_id])
+    ips = response["SecurityGroups"][0]["IpPermissions"]
+    target = next(filter(lambda x: x.get("FromPort", -1) == port, ips), {})
+    assert_that(target["IpRanges"][0]["CidrIp"]).is_equal_to(expected)
+
+
+@pytest.mark.regions(["eu-west-1"])
+@pytest.mark.oss(["centos7"])
+@pytest.mark.schedulers(["sge"])
+def test_dcv_connection(region, instance, os, scheduler, pcluster_config_reader, clusters_factory, test_datadir):
+    remote_command_executor = _setup_cluster(clusters_factory, pcluster_config_reader)
+    _assert_dcv_authenticator_blocking(remote_command_executor)
+    _assert_dcv_working("8444", remote_command_executor)
+
+
+@pytest.mark.regions(["eu-west-1"])
+@pytest.mark.oss(["centos7"])
+@pytest.mark.schedulers(["sge"])
+def test_custom_dcv_configuration(
```

Comment:
you could try to change the "shared_dir" parameter and verify if it is correctly set in dcv configuration since we are using it